### ISSUE TITLE
Exclude specific grpc-client call from being instrumented 

### ISSUE THREAD

**Issue by viktorvoltaire** [DataDog/dd-trace-java#2467]
Hi, 

Is there any way to exclude a specific grpc-client call from being instrumented, our Google Pub/Sub Subscribers are generating 100% errors due to streaming pull (which is expected). You can see more here:

https://cloud.google.com/pubsub/docs/pull#streamingpull_has_a_100_error_rate_this_is_to_be_expected

I have tried the https://docs.datadoghq.com/tracing/setup_overview/configure_data_security/?tab=mongodb#exclude-resources-from-being-collected but since it doesn't seem to work for this.

**Comment by richardstartin** [comment]
Hi @viktorvoltaire we don't currently support this but I think it would be better to filter these out as soon as possible, so do it in the instrumentation. I think it would be better for us to figure out that there is a streaming pull in process and either not trace it, or ignore the error code. Since this will be effecting billing (we retain all errors) I will try to get this prioritised.

**Comment by viktorvoltaire** [comment]
> Hi @viktorvoltaire we don't currently support this but I think it would be better to filter these out as soon as possible, so do it in the instrumentation. I think it would be better for us to figure out that there is a streaming pull in process and either not trace it, or ignore the error code. Since this will be effecting billing (we retain all errors) I will try to get this prioritised.

Thank you. 

For us I think not tracing it would be the goto solution, looking at the actual trace being generated by this right now. It can last up to 20 minutes, meaning the flame graph gets very hard to read and the trace can include multiple thousand of spans

**Comment by richardstartin** [comment]
@viktorvoltaire I think #2482 will fix this, but I need to figure out a way to test it before I can merge it, and whether we need this to be behind a feature flag or on by default. If you're open to trying out a pre-release build, I would find it helpful if you tested the build [here on circleci](https://134515-89221572-gh.circle-artifacts.com/0/libs/dd-java-agent-0.76.0-SNAPSHOT.jar)

**Comment by viktorvoltaire** [comment]
@richardstartin Sure, I will give it a try and come back to you

**Comment by richardstartin** [comment]
I think the solution is too coarse grained and will likely disable tracing of some things you would like to trace. I think a better solution would be to allow you to configure a list of RPC names you want to ignore as a comma separated list.

So if you have a service like this, and want to ignore `Conversation` which is streaming

```proto
syntax = "proto3";

package example;

service Greeter {
  rpc SayHello (Request) returns (Response) {
  }

  rpc Conversation (stream Response) returns (stream Response) {
  }
}

message Request {
  string name = 1;
}

message Response {
  string message = 1;
}
```

you'd set `-Ddd.trace.grpc.ignored.methods=example.Greeter/Conversation` or `DD_TRACE_GRPC_IGNORED_METHODS=example.Greeter/Conversation`

Unfortunately this would put it on you to figure out the names of the streaming RPCs you want to exclude by looking at proto files which may be within the libraries you're using, but does this sound like a workable solution?

**Comment by viktorvoltaire** [comment]
I tried the pre-release, and it seems to do the job.

However excluding specific methods would be the best way forward IMO, its also easy to find the RPC by just looking at the resource name in the trace list

**Comment by richardstartin** [comment]
Thanks, we will proceed with the excluded methods approach to give finer control. I aim to get this into our next release.

**Comment by viktorvoltaire** [comment]
> Thanks, we will proceed with the excluded methods approach to give finer control. I aim to get this into our next release.

Awesome thanks @richardstartin

**Comment by richardstartin** [comment]
#2482 is now merged and will be released in 0.76.0 next week. I changed the property name to leave it open to implement exclusion of inbound calls (e.g. health endpoints), so you'd have to set:

`DD_TRACE_GRPC_IGNORED_OUTBOUND_METHODS=google.pubsub.v1.Subscriber/StreamingPull`

**Comment by viktorvoltaire** [comment]
> #2482 is now merged and will be released in 0.76.0 next week. I changed the property name to leave it open to implement exclusion of inbound calls (e.g. health endpoints), so you'd have to set:
> 
> `DD_TRACE_GRPC_IGNORED_OUTBOUND_METHODS=google.pubsub.v1.Subscriber/StreamingPull`

Excellent 👍  @richardstartin
